src/core/types.h

```diff
@@ -788,7 +788,8 @@ int binarySearch(T needle, RandomAccessIterator start, RandomAccessIterator end,
     int l = 0;
     int r = end - start - 1;
     while (l <= r) {
-        int mid = l + (r - l) / 2;
```

Comment:
@Daetalus l + (r - l) can be overflow if needle value is right-end near value,  l and r can be large
how about read this article 

http://googleresearch.blogspot.kr/2006/06/extra-extra-read-all-about-it-nearly.html